src/main/java/com/google/sps/servlets/DirectionsServlet.java

```diff
@@ -0,0 +1,95 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.common.collect.ImmutableList;
+import com.google.gson.Gson;
+import com.google.sps.exceptions.DirectionsException;
+import com.google.sps.model.DirectionsClient;
+import com.google.sps.model.DirectionsClientFactory;
+import com.google.sps.model.DirectionsClientImpl;
+import com.google.sps.utility.KeyProvider;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Serves key information from optimizing between addresses. */
+@WebServlet("/directions")
+public class DirectionsServlet extends HttpServlet {
+
+  private final DirectionsClientFactory directionsClientFactory;
+  private final String apiKey;
+  private final String origin;
+  private final String destination;
+  private final List<String> waypoints;
+
+  /**
+   * Construct servlet with default DirectionsClient.
+   *
+   * @throws IOException
+   */
+  public DirectionsServlet() throws IOException {
+    directionsClientFactory = new DirectionsClientImpl.Factory();
+    apiKey = (new KeyProvider()).getKey("apiKey");
+    origin = "Waterloo, ON";
+    destination = "Waterloo, ON";
+    waypoints = ImmutableList.of("Montreal, QC", "Windsor, ON", "Kitchener, ON");
+  }
+
+  /**
+   * Construct servlet with explicit implementation of DirectionsClient.
+   *
+   * @param factory A DirectionsClientFactory containing the implementation of
+   *     DirectionsClientFactory.
+   */
+  public DirectionsServlet(
+      DirectionsClientFactory factory,
+      String fakeApiKey,
+      String fakeOrigin,
+      String fakeDestination,
+      List<String> fakeWaypoints) {
+    directionsClientFactory = factory;
+    apiKey = fakeApiKey;
+    origin = fakeOrigin;
+    destination = fakeDestination;
+    waypoints = fakeWaypoints;
+  }
+
+  /**
+   * Returns the most optimal order of travel between addresses.
+   *
+   * @param request HTTP request from the client.
+   * @param response HTTP response to the client.
+   * @throws ServletException
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException {
+    try {
+      DirectionsClient directionsClient = directionsClientFactory.getDirectionsClient(apiKey);
+      List<String> directions = directionsClient.getDirections(origin, destination, waypoints);
+      Gson gson = new Gson();
+      String directionsJson = gson.toJson(directions);
+      response.setContentType("application/json");
+      response.getWriter().println(directionsJson);
```

Comment:
Use `JsonUtil`?

Reply:
Absolutely, I've changed this in this PR and will do the same for all calls made prior to JsonUtility being merged in another PR.

Reply:
`RESOLVED`